### PR TITLE
Update gradle and lint to latest stable

### PR DESCRIPTION
**Pull request by ZacSweers**
Opting to baseline the existing test mode issues (see the doc on that new `open val testModes` prop for details) to fix in subsequent PRs

### CODE REVIEW

File: slack-lint/build.gradle.kts [hunk 1]
```diff
@@ -18,43 +18,36 @@ import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
...
 dependencies {
-  compileOnly("com.android.tools.lint:lint-api:$lintVersion")
-  compileOnly("com.android.tools.lint:lint-checks:$lintVersion")
-  ksp("dev.zacsweers.autoservice:auto-service-ksp:1.0.0")
-  implementation("com.google.auto.service:auto-service-annotations:1.0")
-  testImplementation("com.android.tools.lint:lint:$lintVersion")
-  testImplementation("com.android.tools.lint:lint-tests:$lintVersion")
-  testImplementation("com.android.tools:testutils:$lintVersion")
-  testImplementation("junit:junit:4.13.2")
+  compileOnly(libs.bundles.lintApi)
```
Comment (seankim-android) [inline slack-lint/build.gradle.kts:39]: TIL bundles

**Review by seankim-android — APPROVED** [review]
lgtm!

**Review by adamsp — APPROVED** [review]

File: slack-lint/src/test/java/slack/lint/BaseSlackLintTest.kt [hunk 2]
```diff
@@ -42,6 +43,8 @@ abstract class BaseSlackLintTest : LintDetectorTest() {
   /** Optional override to customize the lint client name when running lint test tasks. */
   open val lintClientName: String? = null
 
+  open val skipTestModes: Array<TestMode>? = null
```
[the remark this answers is not in the document]
Comment (ZacSweers) [reply slack-lint/src/test/java/slack/lint/BaseSlackLintTest.kt:46]: Well this is awkward, what happened to my doc??

[the remark this answers is not in the document]
Comment (ZacSweers) [reply slack-lint/src/test/java/slack/lint/BaseSlackLintTest.kt:54]: [e67e01f](https://github.com/slackhq/slack-lints/pull/19/commits/e67e01f58736255c06409480c7dc3732325874bf)

**Pull request merged**